batcher/aligned-batcher/src/lib.rs

```diff
@@ -1167,39 +1178,83 @@ impl Batcher {
         proof_submitters: Vec<Address>,
         fee_params: CreateNewTaskFeeParams,
     ) -> Result<TransactionReceipt, BatcherError> {
-        info!("Creating task for: 0x{}", hex::encode(batch_merkle_root));
-
-        match try_create_new_task(
-            batch_merkle_root,
-            batch_data_pointer.clone(),
-            proof_submitters.clone(),
-            fee_params.clone(),
-            &self.payment_service,
-        )
-        .await
-        {
-            Ok(receipt) => Ok(receipt),
-            Err(BatcherSendError::TransactionReverted(err)) => {
-                // Since transaction was reverted, we don't want to retry with fallback.
-                warn!("Transaction reverted {:?}", err);
-
-                Err(BatcherError::TransactionSendError)
-            }
-            Err(_) => {
-                let receipt = try_create_new_task(
+        let result = retry_function(
+            || {
+                create_new_task_retryable(
```

Comment:
Are we not bumping the `gasPrice` here?

Reply:
We can't bump the task since we are handling the user's fees here.

Reply:
Right